In [369]:
import pandas as pd
import numpy as np

Pour reference:
http://www.fao.org/faostat/fr/#data

# Question 1

Donnez le résultat de votre calcul pour l'année 2013, ainsi que pour la dernière année  disponible au jour où vous effectuez ce projet.

In [370]:
# Loading csv file
population = pd.read_csv("population_world_2013.csv")


In [371]:
# Rebranding significant columns
population.columns = ["xx1", "source", "id", "pays", "code", "element", "xx3", "type", "year", "yearbis", "unit", "value", "xx6", "xx7"]
#population.set_index('pays', inplace=True)
population.loc[population["pays"]=="France"]

xx1               source  id    pays  code            element   xx3  \
58  FBS  Bilans Alimentaires  68  France   511  Population totale  2501   

          type  year  yearbis            unit  value  xx6                xx7  
58  Population  2013     2013  1000 personnes  64291  NaN  Donnée officielle

In [372]:
population.shape

(175, 14)

In [373]:
# Pivoting and filtering on the value column
world_population = population.pivot_table(
    index=["pays","id", "year"], 
    #columns=["element"],
    values=["value"]
)

first_estimate = world_population.sum()*1000
first_estimate

value    8413993000
dtype: int64

In [374]:
# Aggregation_chine
world_population.loc[["Chine, continentale", "Chine, Taiwan Province de", "Chine - RAS de Macao", "Chine - RAS de Hong-Kong"], 'value'].sum()

1416667

On obtient ainsi une estimation de la population mondiale pour l'annee 2013 a plus de 8,4 milliards.

Ce chiffre est tres largement superieur aux estimations de l'ONU pour l'annee 2013 qui se situent a 7 milliards pour l'annee 2013. Par ailleurs pour l'annee 2019, l'estimation actuelle est de 7,7 milliards (source Ined https://www.ined.fr/fr/tout-savoir-population/chiffres/tous-les-pays-du-monde/)

Notre calcul est donc bien superieur a l'estimation. En parcourant la liste des noms de pays, on remarque que la Chine apparait a plusieurs reprises selon differentes declinaisons. 
Ici l'unite des chiffres utilises est une unite par millier de personnes (marge d'erreur max de 2%) 
La source des donnees pour la Chine est issue d'aggregat de donnees et pese plus de 16% de la population mondiale. 
Certaines donnees sont manquantes (URSS, Coree du Nord).


In [375]:
# Aggregation_chine
china_pop = world_population.loc["Chine", "value"].sum() * 1000
china_pop

1416667000

Les donnees concernant la population chinoise sont en doublon. Aussi il ne faut retenir que la valeur globale. On retrouve alors un calcul plus proche de l'estimation qui est a 7 milliards.

In [376]:
world_population = first_estimate - china_pop
world_population.value

6997326000

###### On se rapproche ainsi de l'estimation globale pour l'annee 2013.
> **_Reponse 1_**: 6 997 326 000 personnes


---------------------------------------

# Question 2
Identifiez ces redondances, en donnant votre réponse sous forme de formule mathématique (pas besoin de coder ici). C'est une équation à 3 termes de type \(a_1 + a2 + [...] = b_1 + b_2 + [...] = c_1 + c_2 + [...]\) ) faisant intervenir chacune des 11 quantités données ci dessus. Illustrez cette équation avec l'exemple du blé en France. Pour avoir un indice, cliquez sur "Définitions et Standards" sur la page de téléchargement des données.
* Production (Production)
* Importations (Import Quantity)
* Exportations (Export Quantity)
* Variation de stock (Stock Variation)
* Disponibilité intérieure (Domestic supply)
* Semences (Seed)
* Pertes (Waste)
* Nourriture (Food) , aussi appelée Disponibilité alimentaire (Food Supply)
* Aliments pour animaux (Feed)
* Traitement (Processing)
* Autres utilisations (Other uses)


In [377]:
vegetal = pd.read_csv("veg.csv")
vegetal.columns = ["xx1", "source", "id", "pays", "code", "element", "xx3", "type", "year", "yearbis", "unit", "value", "xx6", "xx7"]

# Global
# vegetal

# Focus on France
vegetal.loc[vegetal["pays"]=="France"].head(100)

xx1               source  id    pays  code  \
30989  FBS  Food Balance Sheets  68  France  5511   
30990  FBS  Food Balance Sheets  68  France  5611   
30991  FBS  Food Balance Sheets  68  France  5072   
30992  FBS  Food Balance Sheets  68  France  5911   
30993  FBS  Food Balance Sheets  68  France  5301   
30994  FBS  Food Balance Sheets  68  France  5521   
30995  FBS  Food Balance Sheets  68  France  5527   
30996  FBS  Food Balance Sheets  68  France  5123   
30997  FBS  Food Balance Sheets  68  France  5131   
30998  FBS  Food Balance Sheets  68  France  5154   
30999  FBS  Food Balance Sheets  68  France  5142   
31000  FBS  Food Balance Sheets  68  France   645   
31001  FBS  Food Balance Sheets  68  France   664   
31002  FBS  Food Balance Sheets  68  France   674   
31003  FBS  Food Balance Sheets  68  France  5511   
31004  FBS  Food Balance Sheets  68  France  5611   
31005  FBS  Food Balance Sheets  68  France  5072   
31006  FBS  Food Balance Sheets  68  France  5911   
31007  FBS  Food Balance Sheets  68  France  5301   
31008  FBS  Food Balance Sheets  68  France  5521   
31009  FBS  Food Balance Sheets  68  France  5527   
31010  FBS  Food Balance Sheets  68  France  5123   
31011  FBS  Food Balance Sheets  68  France  5131   
31012  FBS  Food Balance Sheets  68  France  5154   
31013  FBS  Food Balance Sheets  68  France  5142   
31014  FBS  Food Balance Sheets  68  France   645   
31015  FBS  Food Balance Sheets  68  France   664   
31016  FBS  Food Balance Sheets  68  France   674   
31017  FBS  Food Balance Sheets  68  France  5511   
31018  FBS  Food Balance Sheets  68  France  5611   
...    ...                  ...  ..     ...   ...   
31059  FBS  Food Balance Sheets  68  France  5521   
31060  FBS  Food Balance Sheets  68  France  5527   
31061  FBS  Food Balance Sheets  68  France  5123   
31062  FBS  Food Balance Sheets  68  France  5142   
31063  FBS  Food Balance Sheets  68  France   645   
31064  FBS  Food Balance Sheets  68  France   664   
31065  FBS  Food Balance Sheets  68  France   674   
31066  FBS  Food Balance Sheets  68  France  5511   
31067  FBS  Food Balance Sheets  68  France  5611   
31068  FBS  Food Balance Sheets  68  France  5911   
31069  FBS  Food Balance Sheets  68  France  5301   
31070  FBS  Food Balance Sheets  68  France  5521   
31071  FBS  Food Balance Sheets  68  France  5527   
31072  FBS  Food Balance Sheets  68  France  5511   
31073  FBS  Food Balance Sheets  68  France  5611   
31074  FBS  Food Balance Sheets  68  France  5072   
31075  FBS  Food Balance Sheets  68  France  5911   
31076  FBS  Food Balance Sheets  68  France  5301   
31077  FBS  Food Balance Sheets  68  France  5521   
31078  FBS  Food Balance Sheets  68  France  5527   
31079  FBS  Food Balance Sheets  68  France  5511   
31080  FBS  Food Balance Sheets  68  France  5611   
31081  FBS  Food Balance Sheets  68  France  5072   
31082  FBS  Food Balance Sheets  68  France  5911   
31083  FBS  Food Balance Sheets  68  France  5301   
31084  FBS  Food Balance Sheets  68  France  5521   
31085  FBS  Food Balance Sheets  68  France  5527   
31086  FBS  Food Balance Sheets  68  France  5131   
31087  FBS  Food Balance Sheets  68  France  5142   
31088  FBS  Food Balance Sheets  68  France   645   

                                      element   xx3                      type  \
30989                              Production  2511        Wheat and products   
30990                         Import Quantity  2511        Wheat and products   
30991                         Stock Variation  2511        Wheat and products   
30992                         Export Quantity  2511        Wheat and products   
30993                Domestic supply quantity  2511        Wheat and products   
30994                                    Feed  2511        Wheat and products   
30995                                    Seed  2511        Wheat and products   
30996                                  Losses  2511        Wheat and pro

Au regard des chiffres de la France pour le ble, on intuite les relations suivantes:
* Domestic Supply = Production + Import Quantity - Export Quantity + Stock variation
* Domestic Supply = Food + Feed + Seed + Losses + Other Uses + Processing

In [378]:
vegetal.set_index(['pays', 'element', 'type'], inplace=True)
vegetal.shape

(96715, 11)

In [379]:
# Hypothesis check
production = vegetal.loc['France', 'Production', 'Wheat and products'].value
import_quantity = vegetal.loc['France', 'Import Quantity', 'Wheat and products'].value
stock_variation = vegetal.loc['France', 'Stock Variation', 'Wheat and products'].value
export_quantity = vegetal.loc['France', 'Export Quantity', 'Wheat and products'].value

food = vegetal.loc['France', 'Food', 'Wheat and products'].value
feed = vegetal.loc['France', 'Feed', 'Wheat and products'].value
seed = vegetal.loc['France', 'Seed', 'Wheat and products'].value
losses = vegetal.loc['France', 'Losses', 'Wheat and products'].value
other_uses = vegetal.loc['France', 'Other uses', 'Wheat and products'].value
processing = vegetal.loc['France', 'Processing', 'Wheat and products'].value

domestic_supply = vegetal.loc['France', 'Domestic supply quantity', 'Wheat and products'].value

In [380]:
# First check
domestic_supply - (food + feed + seed + losses + other_uses + processing)

0.0

In [381]:
# Second check
domestic_supply - (production + import_quantity - export_quantity + stock_variation)

0.0

> **_Reponse 2_**: On en conclut que:
> Domestic supply = feed + seed + losses + other_uses + processing = production + import_quantity - export_quantity + stock_variation


--------------------

# Question 3
Calculez (pour chaque pays et chaque produit) la disponibilité alimentaire en kcal puis en kg de protéines.

Vous ferez cela à partir de ces informations :

* Population de chaque pays
* Disponibilité alimentaire ("Food Supply" en anglais) donnée pour chaque produit et pour chaque pays en kcal/personne/jour.
* Disponibilité alimentaire en protéines ("Protein supply quantity" en anglais) donnée pour chaque produit et pour chaque pays en g/personne/jour.

In [382]:
# Making vegetal and animal food table
animal = pd.read_csv("ani.csv")
vegetal = pd.read_csv("veg.csv")
animal['origin']='animal'
vegetal['origin']='vegetal'

temp=animal.append(vegetal)

In [383]:
temp.columns = ["xx","xx2","country_code","country",'xx3','element'
    ,'item_code','item','xx4',"year","unit","value",'xx5','xx6'
    ,'origin']

In [384]:
food = temp.pivot_table(
    index = ["country_code","country","item_code","item","year","origin"],
    columns = ["element"], 
    values=["value"], 
    aggfunc=sum,
    fill_value=0
)
food.columns = ['domestic_supply_quantity', 'export_quantity', 'feed',
    'food', 'food_supply_kcalcapitaday', 
    'food_supply_quantity_kgcapitayr', 'import_quantity', 'other_uses',
    'processing', 'production', 'protein_supply_quantity_gcapitaday', 
    'seed', 'stock_variation', 'waste']

food = food.reset_index()
food.head()

country_code  country  item_code                 item  year   origin  \
0             1  Armenia       2511   Wheat and products  2013  vegetal   
1             1  Armenia       2513  Barley and products  2013  vegetal   
2             1  Armenia       2514   Maize and products  2013  vegetal   
3             1  Armenia       2515     Rye and products  2013  vegetal   
4             1  Armenia       2516                 Oats  2013  vegetal   

   domestic_supply_quantity  export_quantity  feed  food  \
0                       554                1    93   389   
1                       198                0   137     0   
2                       102                0    96     0   
3                         1                0     1     0   
4                         6                0     4     1   

   food_supply_kcalcapitaday  food_supply_quantity_kgcapitayr  \
0                       1024                           130.60   
1                          0                             0.00   
2                          0                             0.03   
3                          1                             0.12   
4                          2                             0.37   

   import_quantity  other_uses  processing  production  \
0              361          32           0          10   
1                9          15          26           7   
2               82           7           0           0   
3                0           0           0           0   
4                1           0           0           0   

   protein_supply_quantity_gcapitaday   seed  stock_variation  waste  
0                                 312  30.52               30   -118  
1                                 189   0.00               14      0  
2                                  21   0.01                0      0  
3                                   1   0.02                0      0  
4                                   5   0.09                0      0

In [385]:
# Making new population table 
population = pd.read_csv("population_world_2013.csv")
population.columns = ["xx","xx2","country_code","country",'xx3','element'
    ,'item_code','item','xx4',"year","unit","value",'xx5','xx6']
population = population.pivot_table(
    index = ["country_code", "country", "item_code", "item", "year"],
    columns = ["element"],
    values = ["value"],
    aggfunc=sum,
    fill_value=0
)
population.reset_index()
population.head()

value
element                                            Population totale
country_code country     item_code item       year                  
1            Arménie     2501      Population 2013              2977
2            Afghanistan 2501      Population 2013             30552
3            Albanie     2501      Population 2013              3173
4            Algérie     2501      Population 2013             39208
7            Angola      2501      Population 2013             21472

In [386]:
# Merging tables on column 'country_code'
df_tmp = pd.merge(population, food, on="country_code", left_index=True)
df_tmp

c:\users\hieu.truong\.virtualenvs\ocr_da_proj3-07-v1_mv\lib\site-packages\pandas\core\reshape\merge.py:522: UserWarning: merging between different levels can give an unintended result (2 levels on the left, 1 on the right)
  warnings.warn(msg, UserWarning)


country_code  (value, Population totale)  country  item_code  \
0                 1                        2977  Armenia       2511   
1                 1                        2977  Armenia       2513   
2                 1                        2977  Armenia       2514   
3                 1                        2977  Armenia       2515   
4                 1                        2977  Armenia       2516   
5                 1                        2977  Armenia       2517   
6                 1                        2977  Armenia       2518   
7                 1                        2977  Armenia       2520   
8                 1                        2977  Armenia       2531   
9                 1                        2977  Armenia       2532   
10                1                        2977  Armenia       2534   
11                1                        2977  Armenia       2537   
12                1                        2977  Armenia       2542   
13                1                        2977  Armenia       2543   
14                1                        2977  Armenia       2546   
15                1                        2977  Armenia       2547   
16                1                        2977  Armenia       2549   
17                1                        2977  Armenia       2551   
18                1                        2977  Armenia       2555   
19                1                        2977  Armenia       2556   
20                1                        2977  Armenia       2557   
21                1                        2977  Armenia       2558   
22                1                        2977  Armenia       2560   
23                1                        2977  Armenia       2561   
24                1                        2977  Armenia       2563   
25                1                        2977  Armenia       2570   
26                1                        2977  Armenia       2571   
27                1                        2977  Armenia       2572   
28                1                        2977  Armenia       2573   
29                1                        2977  Armenia       2574   
...             ...                         ...      ...        ...   
15672           351                     1416667    China       2656   
15673           351                     1416667    China       2657   
15674           351                     1416667    China       2658   
15675           351                     1416667    China       2659   
15676           351                     1416667    China       2680   
15677           351                     1416667    China       2731   
15678           351                     1416667    China       2732   
15679           351                     1416667    China       2733   
15680           351                     1416667    China       2734   
15681           351                     1416667    China       2735   
15682           351                     1416667    China       2736   
15683           351                     1416667    China       2737   
15684           351                     1416667    China       2740   
15685           351                     1416667    China       2743   
15686           351                     1416667    China       2744   
15687           351                     1416667    China       2745   
15688           351                     1416667    China       2761   
15689           351                     1416667    China       2762   
15690           351                     1416667    China       2763   
15691           351                     1416667    China       2764   
15692           351                     1416667    China       2765   
15693           351                     1416667    China       2766   
15694           351                     1416667    China       2767   
15695           351                     1416667    China       2769   
15696           351            

In [387]:
result = df_tmp.loc[:, 
    [
        "country", 
        "item",
        ('value', 'Population totale'), 
        "food",
        "feed",
        "waste",
        "origin",
        "domestic_supply_quantity",
        'food_supply_kcalcapitaday', 
        'protein_supply_quantity_gcapitaday'
    ]
]
result.head()


country                 item  (value, Population totale)  food  feed  \
0  Armenia   Wheat and products                        2977   389    93   
1  Armenia  Barley and products                        2977     0   137   
2  Armenia   Maize and products                        2977     0    96   
3  Armenia     Rye and products                        2977     0     1   
4  Armenia                 Oats                        2977     1     4   

   waste   origin  domestic_supply_quantity  food_supply_kcalcapitaday  \
0   -118  vegetal                       554                       1024   
1      0  vegetal                       198                          0   
2      0  vegetal                       102                          0   
3      0  vegetal                         1                          1   
4      0  vegetal                         6                          2   

   protein_supply_quantity_gcapitaday  
0                                 312  
1                                 189  
2                                  21  
3                                   1  
4                                   5

In [388]:
result["food_supply_kcalayear"] = result[('value', 'Population totale')] * 1000 * result["food_supply_kcalcapitaday"] * 365
result["protein_supply_quantity_kgayear"] = (result[('value', 'Population totale')] * 1000 * result["protein_supply_quantity_gcapitaday"] * 365) / 1000

In [389]:
q3_result = result.loc[:, ["country", "item", "food_supply_kcalayear", "protein_supply_quantity_kgayear", "food"]]
q3_result.head()

country                 item  food_supply_kcalayear  \
0  Armenia   Wheat and products          1112683520000   
1  Armenia  Barley and products                      0   
2  Armenia   Maize and products                      0   
3  Armenia     Rye and products             1086605000   
4  Armenia                 Oats             2173210000   

   protein_supply_quantity_kgayear  food  
0                      339020760.0   389  
1                      205368345.0     0  
2                       22818705.0     0  
3                        1086605.0     0  
4                        5433025.0     1

---------------------------

# Question 4
A partir de ces dernières informations, et à partir du poids de la disponibilité alimentaire (pour chaque pays et chaque produit), calculez pour chaque produit le ratio "énergie/poids", que vous donnerez en kcal/kg. Vous pouvez vérifier la cohérence de votre calcul en comparant ce ratio aux données disponibles sur internet, par exemple en cherchant la valeur calorique d'un oeuf.

>Indication : La disponibilité alimentaire en kcal/personne/jour est calculée par la FAO en multipliant la quantité Nouriture (Food) par le ratio énergie/poids (en kcal/kg), puis en le divisant par la population du pays puis par 365. Ici, on vous demande juste de retrouver le ratio énergie/poids que la FAO a utilisé dans son calcul.

En suivant la même méthodologie, calculez également le pourcentage de protéines de chaque produit (pour chaque pays). Ce pourcentage est obtenu en calculant le ratio "poids de protéines/poids total" (attention aux unités utilisées). Vous pouvez vérifier la cohérence de votre calcul en comparant ce ratio aux données disponibles sur internet, par exemple en cherchant la teneur en protéines de l'avoine.

https://www.healthline.com/health/food-nutrition/calories-in-an-egg
According to the United States Department of Agriculture (USDA), there are about 72 caloriesTrusted Source in a large egg. A large egg weighs 50 grams (g).

soit une moyenne de 1440 kcal/kg

https://oeuf.ca/faq/valeur-nutritive-dun-oeuf/
Un gros œuf (53 g) de catégorie A contient 6 g de protéines et seulement 70 calories

soit une moyenne de 1320 kcal/kg

https://fr.wikipedia.org/wiki/%C5%92uf_(aliment)#Valeur_nutritionnelle
615 kJ (147 kcal) pour 100g

soit une moyenne de 1470 kcal/kg

In [390]:
result["ratio_kcalakg"] = (result["food_supply_kcalcapitaday"] * result[('value', 'Population totale')] * 1000 * 365) / (result["food"] * 1000000)
result.loc[~np.isfinite(result['ratio_kcalakg']), 'ratio_kcalakg'] = np.nan
q4_result = result

In [391]:
result.query('item=="Eggs"')["ratio_kcalakg"].mean()

1346.02239399714

**_Reponse 4-1_**: On trouve un resultat proche des estimations remontees ci-dessus 1346 kcal/kg


In [392]:
result.columns

Index([                           'country',
                                     'item',
             ('value', 'Population totale'),
                                     'food',
                                     'feed',
                                    'waste',
                                   'origin',
                 'domestic_supply_quantity',
                'food_supply_kcalcapitaday',
       'protein_supply_quantity_gcapitaday',
                    'food_supply_kcalayear',
          'protein_supply_quantity_kgayear',
                            'ratio_kcalakg'],
      dtype='object')

In [393]:
result["ratio_protakg"] = (result["protein_supply_quantity_gcapitaday"]  * result[('value', 'Population totale')] * 1000 * 365) / (result["food"] * 1000000000)
tmp = result[["country", "item", "ratio_protakg"]]

In [394]:
result.loc[~np.isfinite(result['ratio_protakg']), 'ratio_protakg'] = np.nan
result.query('item=="Oats"')["ratio_protakg"].mean(skipna=True)

174.66637564891477

https://en.wikipedia.org/wiki/Oat
169 g per kg soit 16,9% de proteine pour l'avoine

--------

# Question 5
Citez 5 aliments parmi les 20 aliments les plus caloriques, en utilisant le ratio énergie/poids. 
Étonnamment, il arrive que ce ratio soit différent en fonction du pays. Il faudra donc réaliser pour chaque aliment une moyenne sur les différents pays. Vous créerez donc une nouvelle table grâce à une agrégation. Attention à bien retirer les valeurs égales à 0 afin de ne pas fausser le calcul de la moyenne.
Citez 5 aliments parmi les 20 aliments les plus riches en protéines.

In [395]:
q4_result.head()

country                 item  (value, Population totale)  food  feed  \
0  Armenia   Wheat and products                        2977   389    93   
1  Armenia  Barley and products                        2977     0   137   
2  Armenia   Maize and products                        2977     0    96   
3  Armenia     Rye and products                        2977     0     1   
4  Armenia                 Oats                        2977     1     4   

   waste   origin  domestic_supply_quantity  food_supply_kcalcapitaday  \
0   -118  vegetal                       554                       1024   
1      0  vegetal                       198                          0   
2      0  vegetal                       102                          0   
3      0  vegetal                         1                          1   
4      0  vegetal                         6                          2   

   protein_supply_quantity_gcapitaday  food_supply_kcalayear  \
0                                 312          1112683520000   
1                                 189                      0   
2                                  21                      0   
3                                   1             1086605000   
4                                   5             2173210000   

   protein_supply_quantity_kgayear  ratio_kcalakg  ratio_protakg  
0                      339020760.0    2860.368946       0.871519  
1                      205368345.0            NaN            NaN  
2                       22818705.0            NaN            NaN  
3                        1086605.0            NaN            NaN  
4                        5433025.0    2173.210000       5.433025

In [396]:
q5_result = q4_result.groupby('item').agg("mean").sort_values('ratio_kcalakg', ascending=False)

In [397]:
q5_result[['ratio_kcalakg']].head(20)

ratio_kcalakg
item                                  
Fish, Body Oil             9003.715504
Palmkernel Oil             8826.979763
Ricebran Oil               8727.700518
Palm Oil                   8721.172690
Maize Germ Oil             8662.278512
Oilcrops Oil, Other        8630.601679
Cottonseed Oil             8612.988561
Soyabean Oil               8573.634352
Rape and Mustard Oil       8503.550297
Olive Oil                  8465.878436
Sunflowerseed Oil          8432.286866
Groundnut Oil              8389.050961
Coconut Oil                8326.942490
Sesameseed Oil             7887.763285
Fats, Animals, Raw         7578.965784
Butter, Ghee               7343.215549
Groundnuts (Shelled Eq)    5617.029957
Sesame seed                5476.527916
Fish, Liver Oil            4991.466250
Palm kernels               4331.090000

In [398]:
q52_result = q4_result.groupby('item').agg("mean").sort_values('ratio_protakg', ascending=False)

In [399]:
q52_result[['ratio_protakg']].head(20)

ratio_protakg
item                                     
Rape and Mustardseed        143938.816448
Sugar cane                   65381.808956
Oilcrops, Other              49296.066060
Sunflower seed               14507.304637
Soyabeans                    11839.581113
Barley and products           1070.059435
Coffee and products            669.528278
Cephalopods                    570.699417
Maize and products             441.399855
Pulses, Other and products     405.531505
Aquatic Plants                 310.331693
Sorghum and products           237.111679
Oats                           174.666376
Cereals, Other                 170.276062
Palm kernels                   168.410513
Cloves                         159.521439
Groundnuts (Shelled Eq)        146.391283
Ricebran Oil                   145.289177
Rye and products               126.804649
Beans                          117.798889

--------

# Question 6 
Calculez, pour les produits végétaux uniquement, la disponibilité intérieure mondiale exprimée en kcal.

In [400]:
q6_result = q4_result.query('origin=="vegetal"')

In [428]:
q6_result.head()

country                 item  (value, Population totale)  food  feed  \
0  Armenia   Wheat and products                        2977   389    93   
1  Armenia  Barley and products                        2977     0   137   
2  Armenia   Maize and products                        2977     0    96   
3  Armenia     Rye and products                        2977     0     1   
4  Armenia                 Oats                        2977     1     4   

   waste   origin  domestic_supply_quantity  food_supply_kcalcapitaday  \
0   -118  vegetal                       554                       1024   
1      0  vegetal                       198                          0   
2      0  vegetal                       102                          0   
3      0  vegetal                         1                          1   
4      0  vegetal                         6                          2   

   protein_supply_quantity_gcapitaday  food_supply_kcalayear  \
0                                 312          1112683520000   
1                                 189                      0   
2                                  21                      0   
3                                   1             1086605000   
4                                   5             2173210000   

   protein_supply_quantity_kgayear  ratio_kcalakg  ratio_protakg  \
0                      339020760.0    2860.368946       0.871519   
1                      205368345.0            NaN            NaN   
2                       22818705.0            NaN            NaN   
3                        1086605.0            NaN            NaN   
4                        5433025.0    2173.210000       5.433025   

   veg_domestic_supply_energy_all  veg_domestic_supply_prot_all  
0                    1.716221e+12                  5.229112e+08  
1                             NaN                           NaN  
2                             NaN                           NaN  
3                             NaN                           NaN  
4                    1.086605e+10                  2.716512e+07

In [402]:
total_veg_food_kcal = q6_result[['food_supply_kcalayear']].sum().food_supply_kcalayear
total_veg_food_kcal

7284336383795000

--------

# Question 7 
Combien d'humains pourraient être nourris si toute la disponibilité intérieure mondiale de produits végétaux était utilisée pour de la nourriture ? Donnez les résultats en termes de calories, puis de protéines, et exprimez ensuite ces 2 résultats en pourcentage de la population mondiale.

In [403]:
total_veg_food_prot = q6_result[['protein_supply_quantity_kgayear']].sum().protein_supply_quantity_kgayear

In [404]:
total_pop = world_population.value

In [405]:
req_veg_food_kcal = (total_pop * (2940 * 0.5 + 2139 * 0.5) * 365)
req_veg_food_prot = (total_pop * (49 * 0.5 + 41 * 0.5) * 365) * 1000

In [406]:
percent_fed_kcal = (total_veg_food_kcal / req_veg_food_kcal) * 100
percent_fed_kcal

112.30958008143455

https://fr.wikipedia.org/wiki/Prot%C3%A9ine#Quantit%C3%A9s_recommand%C3%A9es
Les besoins moyens en protéines ont été définis par la FAO qui recommande 49 g de protéines pour les hommes adultes et 41 pour les femmes (47 si enceintes, 58,5 si allaitantes).

In [407]:
percent_fed_prot = (total_veg_food_prot / req_veg_food_prot) * 100
percent_fed_prot

2.145054534219373

--------

# Question 8
Combien d'humains pourraient être nourris si toute la disponibilité alimentaire en produits végétaux (Food) la nourriture végétale destinée aux animaux (Feed) et les pertes de produits végétaux (Waste) étaient utilisés pour de la nourriture ? Donnez les résultats en termes de calories, puis de protéines, et exprimez ensuite ces 2 résultats en pourcentage de la population mondiale.

In [429]:
q6_result["veg_domestic_supply_energy_all"] = (q6_result["food"] + q6_result["feed"] + q6_result["waste"]) * q6_result["ratio_kcalakg"] * 1000000
q6_result["veg_domestic_supply_prot_all"] = (q6_result["food"] + q6_result["feed"] + q6_result["waste"]) * q6_result["ratio_protakg"] * 1000000

c:\users\hieu.truong\.virtualenvs\ocr_da_proj3-07-v1_mv\lib\site-packages\ipykernel_launcher.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  """Entry point for launching an IPython kernel.
c:\users\hieu.truong\.virtualenvs\ocr_da_proj3-07-v1_mv\lib\site-packages\ipykernel_launcher.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  


In [430]:
total_veg_food_kcal_all = q6_result[['veg_domestic_supply_energy_all']].sum().veg_domestic_supply_energy_all
total_veg_food_prot_all = q6_result[['veg_domestic_supply_prot_all']].sum().veg_domestic_supply_prot_all

In [431]:
percent_fed_kcal_all = (total_veg_food_kcal_all / req_veg_food_kcal) * 100
percent_fed_kcal_all

157.79678012637447

In [432]:
percent_fed_prot_all = (total_veg_food_prot_all / req_veg_food_prot) * 100
percent_fed_prot_all

53.735359659538794

--------

# Question 9 
Combien d'humains pourraient être nourris avec la disponibilité alimentaire mondiale ? Donnez les résultats en termes de calories, puis de protéines, et exprimez ensuite ces 2 résultats en pourcentage de la population mondiale.

In [433]:
q4_result["domestic_supply_energy_all"] = q4_result["domestic_supply_quantity"] * q4_result["ratio_kcalakg"] * 1000000
q4_result["domestic_supply_prot_all"] = q4_result["domestic_supply_quantity"] * q4_result["ratio_protakg"] * 1000000

In [434]:
total_food_kcal_all = q4_result[['domestic_supply_energy_all']].sum().domestic_supply_energy_all
total_food_prot_all = q4_result[['domestic_supply_prot_all']].sum().domestic_supply_prot_all

In [435]:
percent_fed_kcal = (total_food_kcal_all / req_veg_food_kcal) * 100
percent_fed_kcal

261.6183441796141

In [436]:
percent_fed_prot = (total_food_prot_all / req_veg_food_prot) * 100
percent_fed_prot

477.8940784540712

--------

# Question 10
A partir des données téléchargées qui concernent la sous-nutrition, répondez à cette question : Quelle proportion de la population mondiale est considérée comme étant en sous-nutrition ?

